mangaki/mangaki/views.py

```diff
@@ -411,9 +412,24 @@ def get_profile(request, username):
     user = get_object_or_404(User, username=username)
     is_shared = user.profile.is_shared
     category = request.GET.get('category', 'anime')
+    algo_name = request.GET.get('algo', None)
     ordering = ['favorite', 'willsee', 'like', 'neutral', 'dislike', 'wontsee']
     c = 0
-    rating_list = natsorted(Rating.objects.filter(user__username=username).select_related('work'), key=lambda x: (ordering.index(x.choice), x.work.title.lower()))  # Tri par note puis nom
+    ratings = list(Rating.objects.filter(user__username=username).select_related('work'))
+    compare_function = lambda x: (ordering.index(x.choice), x.work.title.lower())  # By default, sort by rating then name
+    if algo_name is not None:
+        try:
+            work_ids = [rating.work_id for rating in ratings]
+            algo = get_algo_backup(algo_name)
+            dataset = get_dataset_backup(algo_name)
+            best_pos = get_pos_of_best_works_for_user_via_algo(algo, dataset, user.id, work_ids)
+            ranking = {}
+            for rank, pos in enumerate(best_pos):
+                ranking[ratings[pos].id] = rank
+            compare_function = lambda x: (ordering.index(x.choice), ranking[x.id])
```

Comment:
No problem, when looking back ; it was a pure nitpick from the me who likes sophistication. Please disregard this comment :pray:.